core.ts

```diff
@@ -21,42 +30,119 @@ module.exports = function init(config, callback) {
   _.extend(app.settings, { title: 'BRIK' }, config);
   app.config = config;
 
-  process.title = (config.id || "anonymous") + "-chhaang-" + package.version;
+  process.title =
+    (config.id || 'anonymous') + '-chhaang-' + packageJson.version;
 
   app.log = require('logginator')('chhaang', config.log);
-  require('winston-tagged-http-logger')(server, app.log.createSublogger('http'));
+  require('winston-tagged-http-logger')(
+    server,
+    app.log.createSublogger('http')
+  );
+
+  app.use(cookieParser());
+  app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: false }));
 
   // passport & session
   if (config.Feide) {
     var strategy = new SamlStrategy(
       config.Feide.saml || {},
-      function(profile, next) { next(null, profile); });
-    passport.serializeUser(function(user, next) { next(null, user); });
-    passport.deserializeUser(function(user, next) { next(null, user); });
-    passport.use(strategy);
+      function (profile, next) {
+        next(null, profile);
+      }
+    );
+    passport.serializeUser(function (user, next) {
+      next(null, user);
+    });
+    passport.deserializeUser(function (user, next) {
+      next(null, user);
+    });
+    passport.use('saml', strategy);
   }
-  app.use(express.cookieParser());
-  app.use(express.bodyParser());
-  app.use(express.session({ secret: 'sudo apt-get install pants' }));
+  // Open ID
+  else if (config.OpenID) {
+    try {
+      const issuer = await Issuer.discover(config.OpenID.discoveryURL);
+
+      const issuerOptions: ClientMetadata = {
+        client_id: config.OpenID.clientId,
+        client_secret: config.OpenID.clientSecret,
+        redirect_uris: [
+          `${config.OpenID.host}/integration/open-id/login/callback`,
+        ],
+        scope: ['profile'],
+        response_types: ['code'],
+      };
+
+      const client = new issuer.Client({
+        ...issuerOptions,
+      });
+
+      // todo: when we need the access token etc to access third party APIs: https://github.com/panva/node-openid-client#authorization-code-flow
```

Comment:
Right, so in the future if we wanted to call an API from this provider with user we're integrating with, we'd throw this in the database or redis or something at this point? That's worth keeping in mind 👍

Reply:
Yeah. This would be useful for fetching groups in office 365 or the different resources and the users' relations to those resources for Feide!